corehq/apps/app_manager/views/utils.py

```diff
@@ -470,8 +470,17 @@ def handle_shadow_child_modules(app, shadow_parent):
         changes = True
         new_shadow = ShadowModule.new_module(shadow_child.name['en'], 'en')
         new_shadow.source_module_id = shadow_child.unique_id
-        new_shadow.root_module_id = shadow_parent['unique_id']
+        new_shadow.root_module_id = shadow_parent.unique_id
         new_shadow.put_in_root = shadow_child.put_in_root
+
+        # move excluded form ids
```

Comment:
This could be a good helper function that you could then also call from the `create_shadow_children` command.